cpp/src/arrow/compute/kernels/scalar_compare_test.cc

```diff
@@ -1430,6 +1817,262 @@ TYPED_TEST(TestVarArgsCompareParametricTemporal, MaxElementWise) {
+  this->Assert(MaxElementWise, this->scalar(R"("abe")", /*byte_width=*/3),
+               {this->scalar(R"("abe")", /*byte_width=*/3),
+                this->scalar(R"("abc")", /*byte_width=*/3),
+                this->scalar(R"("abd")", /*byte_width=*/3)});
+
+  this->Assert(MaxElementWise,
+               this->array(R"(["abc", "abc", "abd", "abc", "abc"])", /*byte_width=*/3),
+               {this->array(R"(["abc", "abc", "abd", null, "abc"])", /*byte_width=*/3),
+                this->scalar(R"("abc")", /*byte_width=*/3)});
+  this->Assert(MaxElementWise,
+               this->array(R"(["abc", "abc", "abd", "abc", "abc"])", /*byte_width=*/3),
+               {this->array(R"(["abc", null, "abd", null, "abc"])", /*byte_width=*/3),
+                this->scalar(R"("abc")", /*byte_width=*/3)});
+  this->Assert(MaxElementWise,
+               this->array(R"(["abd", "abd", "abd", "abd", "abd"])", /*byte_width=*/3),
+               {this->array(R"(["abc", null, "abd", null, "abc"])", /*byte_width=*/3),
+                this->scalar(R"("abc")", /*byte_width=*/3),
+                this->scalar(R"("abd")", /*byte_width=*/3)});
+  this->Assert(MaxElementWise,
+               this->array(R"(["abc", "abc", "abd", "abc", "abc"])", /*byte_width=*/3),
+               {this->array(R"(["abc", null, "abd", null, "abc"])", /*byte_width=*/3),
+                this->scalar("null", /*byte_width=*/3),
+                this->scalar(R"("abc")", /*byte_width=*/3)});
+
+  // Test null handling
+  this->element_wise_aggregate_options_.skip_nulls = false;
+  this->AssertNullScalar(MaxElementWise, {this->scalar("null"), this->scalar("null")});
+  this->AssertNullScalar(MaxElementWise, {this->scalar(R"("0")"), this->scalar("null")});
+
+  this->Assert(MaxElementWise, this->array(R"(["4", null, "4", "4"])"),
+               {this->array(R"(["1", null, "3", "4"])"), this->scalar(R"("2")"),
+                this->scalar(R"("4")")});
+  this->Assert(MaxElementWise, this->array("[null, null, null, null]"),
+               {this->array(R"(["1", null, "3", "4"])"), this->scalar("null"),
+                this->scalar(R"("2")")});
+  this->Assert(
+      MaxElementWise, this->array(R"(["2", null, "3", "4"])"),
+      {this->array(R"(["1", "2", "3", "4"])"), this->array(R"(["2", null, "2", "2"])")});
+
+  this->Assert(MaxElementWise, this->array("[null, null, null, null]"),
+               {this->scalar(R"("1")"), this->array("[null, null, null, null]")});
+  this->Assert(MaxElementWise, this->array("[null, null, null, null]"),
+               {this->scalar("null"), this->array(R"(["1", "1", "1", "1"])")});
+
+  this->Assert(MaxElementWise,
+               this->array(R"(["abc", "abd", "abd", null, null])", /*byte_width=*/3),
+               {this->array(R"(["abc", "abc", "abd", null, "abc"])", /*byte_width=*/3),
+                this->array(R"(["abc", "abd", "abc", "abc", null])", /*byte_width=*/3)});
+}
+
 TEST(TestMaxElementWiseMinElementWise, CommonTemporal) {
```

Comment:
Can you add a test somewhere that checks an error is returned if input types are different (e.g. different decimals...).